### ISSUE TITLE
Video service occasionally crashes from "Internal data stream error"

### ISSUE THREAD

**Issue by kr3cj** [ivadim/fruitnanny#4]
Thanks @ivadim for making a baby monitor with Raspberry Pi with so little lag. Within the last week, I built fruitnanny on the latest Raspbian Stretch and I've gotten everything working with the latest versions of your software. However, I've noticed that the video service is not reliably starting up all of the time. 

```bash
pi@babycam:~ $ lsb_release -a
No LSB modules are available.
Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 9.3 (stretch)
Release:	9.3
Codename:	stretch
pi@babycam:~ $ uname -a
Linux babycam 4.14.22-v7+ #1096 SMP Mon Feb 26 19:14:22 GMT 2018 armv7l GNU/Linux

pi@babycam:~ $ cat /opt/fruitnanny/bin/video.sh
#!/bin/bash
gst-launch-1.0 -v rpicamsrc name=src preview=0 exposure-mode=night fullscreen=0 bitrate=1000000 annotation-mode=time+date annotation-text-size=20 ! video/x-h264,width=960,height=540,framerate=12/1 ! queue max-size-bytes=0 max-size-buffers=0 ! h264parse ! rtph264pay config-interval=1 pt=96 ! queue ! udpsink host=127.0.0.1 port=5004  sync=false
```
About half the time, the video service won't start up correctly due to an `Internal data stream error` error. Restarting the pi once or twice usually does the trick. Below is the error block in the code:

```bash
mmal: mmal_vc_component_enable: failed to enable component: ENOSPC
/GstPipeline:pipeline0/GstRtpH264Pay:rtph264pay0: seqnum = 7078
ERROR: from element /GstPipeline:pipeline0/GstRpiCamSrc:src: Internal data stream error.
Additional debug info:
gstbasesrc.c(2950): gst_base_src_loop (): /GstPipeline:pipeline0/GstRpiCamSrc:src:
streaming stopped, reason error (-5)
```

I attached 
[syslog.txt](https://github.com/ivadim/fruitnanny/files/1820441/syslog.txt)
 which contains the complete log output from video.sh. You can see that it initially errors but then recovers to start successfully on its own.

Please let me know what else I can do to help.

**Comment by ivadim** [comment]
Hi @kr3cj, 
From error, it's most likely, camera module wasn't ready. Can you please try to add a sleep 5 seconds in video.sh?

**Comment by kr3cj** [comment]
Thanks for the quick reply.

TL;DR
This might have caused some sort of improvement, but only in the eyes of systemd. Browsing to the feed shows a spinning icon for the video.

I tried experimenting with different amounts of sleep, between 5 and 30 seconds. What I see now is that systemd always reports that video.service is up (before, it would report failures from time to time). However, when browsing to the url, I always get a spinning icon for the video feed.

Looking into the logs, sometimes the last log message will report an error similar to the original one reported. Other times, it will report that it started successfully even though I still get a spinning icon after refreshing the webpage. Basically, I don't trust the logs from `sudo journalctl --unit=video.service -n 50` and the video service continues to look unstable.

[log_reports_error_but_videofeed_shows_spinning_icon.txt](https://github.com/ivadim/fruitnanny/files/1823260/log_reports_error_but_videofeed_shows_spinning_icon.txt)

[log_reports_success_but_videofeed_shows_spinning_icon.txt](https://github.com/ivadim/fruitnanny/files/1823257/log_reports_success_but_videofeed_shows_spinning_icon.txt)

**Comment by kr3cj** [comment]
Since janus `v.0.4.2` fixed https://github.com/ivadim/fruitnanny/issues/5, I've done the same to see if it is fixed.

**Comment by ivadim** [comment]
@kr3cj  thank you for the information
